### ISSUE TITLE
Solid-colored concrete / stone bricks

### ISSUE THREAD

**Issue by seeba8** [jpanther/Dectorio#18]
What do you think about solid-colored stone bricks or concrete? I know that there are already mods like that (https://mods.factorio.com/mods/justarandomgeek/color-coding), but I quite like the idea of having the option for colour-coding or pixel-art or so.

**Comment by jpanther** [comment]
This is something I've thought about too. I've been contemplating adding it as a type of painted concrete where you can pick from different colours or also different shapes (like solid or broken lines) that would aid in laying out the factory. I have experimented with the coloured concrete mods in my own factories but never found that it looked right - perhaps due to the washed out colours a lot of these mods are using. The only thing that held me back from doing it so far is that I don't like it when you end up with 50 new items that you need to craft/store/etc just for all the variations.

**Comment by seeba8** [comment]
How about a chest-like machine where you insert any type of concrete and can take out any type of concrete in return? Either through a GUI, making all available at once, or with filter inserters pulling out only the specific ones (you probably need dummy items of everything in order to make that happen), or through the circuit network. You could also make it then have a "locking" option so that it only produces e.g. yellow concrete, and normal inserters pulling out items are enough.
Maybe have some "color" item it uses up, for example made of green chipsets and something, or oil and wood and water. Or a color liquid, for that matter, although I hate liquids.
This would get rid of the storing and creating problem, but still the inventory clutter and the crafting menu clutter persists. I am not sure if there is a way around that.

**Comment by jpanther** [comment]
What I have been thinking is having a coloured/painted concrete item and then when you hold that in your hand you get a GUI that allows you to pick the style. This could work for all different colours, hazard variants, etc and would be very expandable without the need to clutter the crafting menu with dozens of items. It would also mean you only have to carry one type of concrete in your inventory but then have all the options available to you when you come to place it.

**Comment by seeba8** [comment]
True, that's a good point. I don't disagree, but it would make automation (provider chests, roboport, robots building it, or even a personal roboport) impossible, wouldn't it?
Maybe a mix is possible, a carry-on item doing what you described, and a similar feature for an assembly machine or a chest with these outputs that automatically convert into the color taken out?

Maybe it is possible to mark these items as uncraftable (and only make some paint/color item to make it obvious something like this exists)? For example, the item wood is uncraftable, but you can have it in your inventory and receive it from the environment. Or any ore, for that matter. That way, it wouldn't appear in the crafting menu, but you can create it nonetheless.

**Comment by jpanther** [comment]
Some interesting considerations you've raised there too. I don't think it'd be an issue for the logistics network, I guess it depends on play style as you can do all these things many ways.

My original concept for this mod was to build out more of a tech tree so that, for instance, "Paint" would be a technology and you'd then feed that in to unlock the coloured concrete and painted concretes with various assembling chains required. I still want to work towards that, but I also don't want to make it all too complicated. One of the limitations that has slowed me down is that the vanilla game has so few items that make sense for crafting colours or paint so it's about striking a balance that is fun, but also plausible in terms of the reality of manufacturing these things.

**Comment by seeba8** [comment]
100% agree regarding tech tree and resource plausibility. I am looking forward to seeing what you will come up with!
Writing about this really puts me into the mood of trying to write a mod myself. But I am way too uncreative and have no idea for a good mod that does not exist yet. Would be a fun project though, getting into lua and figuring out the factorio modding API.
